docs/src/ref_impl/build_and_run.md

```diff
@@ -159,5 +160,29 @@ In the case of i.MX8, Ghaf deployment contains of creating a bootable SD card wi
 2. To build and flash the Ghaf image:
    1. Run the `nix build .#packages.aarch64-linux.imx8qm-mek-release` command.
    2. Prepare the USB boot media with the target HW image you built: `dd if=./result/nixos.img of=/dev/<YOUR_USB_DRIVE> bs=32M`.
-   
+
 3. Insert an SD card and USB boot media into the board and switch the power on.
+
+---
+
+### Building Ghaf Image for Microchip Icicle kit
+
+Before you begin, check device-independent [prerequisites](./build_and_run.md#prerequisites).
+
+In the case of Icicle kit, Ghaf deployment contains of creating a SD Image which includes UBoot and Linux kernel from Microchip and Ghaf based NixOS rootfs.
+
+1. Build a Ghaf SD image:
+   a. Release Image:
+        Run the `nix build .#packages.riscv64-linux.microchip-icicle-kit-release` command.
+   b. Debug Image:
+        Run the `nix build .#packages.riscv64-linux.microchip-icicle-kit-debug` command.
+
+2. Flash the Ghaf SD image:
+   a. Prepare the SD card with the target HW image you built: `dd if=./result/nixos.img of=/dev/<YOUR_SD_DEVICE> bs=32M`.
+     - Insert an SD card into the board and switch the power on.
+
+   b. You can directly flash nixos image to onboard MMC card: `dd if=./result/nixos.img of=/dev/<YOUR_MMC_DEVICE> bs=32M`.
+     - Please refer Icicle kit user guide for more detail to access MMC card as USB disk.
```

Comment:
Could you please provide the link where the `Icicle Kit user guide` is located?

Then change this `Please refer Icicle Kit user guide for more detail to access the MMC card as a USB disk.` to this:
For more information on how to access the MMC card as a USB disk, see [Icicle Kit user guide]().

It will be just a sentence after the list, no need to make a nested item.

Please keep in mind, that instead of `Icicle Kit user guide` must be a real name for the guide.

Reply:
Done.